pandas/io/excel/_base.py

```diff
@@ -726,7 +726,9 @@ def parse(
 
         output = {}
 
+        last_sheetname = None
```

Comment:
Might be good to check after the loop if `last_sheetname = None` then raise an error that `sheets` was an empty list?

Reply:
Good idea.  What error type would be most appropriate?  My thought is a ValueError, but I wonder whether something else might be better.

Reply:
Yeah ValueError would be good